frontend/src/pages/GraphPF/components/stylesComponentFactory.tsx

```diff
@@ -68,22 +75,94 @@ const nodeContextMenu = (node: GraphElement, kiosk: string): React.ReactElement[
     );
   }
 
+  const contextMenuStyle = kialiStyle({
+    paddingTop: '0.1rem',
+    paddingBottom: '0.1rem'
+  });
+
   const items = optionsPF.map((o, i) => {
     return (
       // TODO: fix kiosk param
       !!o.altClickHandler ? (
-        <ContextMenuItem key={`option-${i}`} onClick={() => o.altClickHandler!(o.node!, kiosk)}>
+        <ContextMenuItem
+          className={contextMenuStyle}
+          key={`option-${i}`}
+          onClick={() => o.altClickHandler!(o.node!, kiosk)}
+        >
           {o.text}
         </ContextMenuItem>
       ) : (
-        <ContextMenuItem key={`option-${i}`} onClick={() => clickHandler(o, kiosk)}>
+        <ContextMenuItem className={contextMenuStyle} key={`option-${i}`} onClick={() => clickHandler(o, kiosk)}>
           {o.text} {o.target === '_blank' && <ExternalLinkAltIcon />}
         </ContextMenuItem>
       )
     );
   });
 
-  return items;
+  if (nodeData.nodeType !== NodeType.SERVICE) {
+    return Promise.resolve(items);
+  }
+
+  let getDetailPromise = API.getServiceDetail(nodeData.namespace, nodeData.service!, false, nodeData.cluster);
+  let getGwPromise = API.getAllIstioConfigs(['gateways'], false, '', '', nodeData.cluster);
+  let getPeerAuthsPromise = API.getIstioConfig(
+    nodeData.namespace,
+    ['peerauthentications'],
+    false,
+    '',
+    '',
+    nodeData.cluster
+  );
+
+  return new Promise<React.ReactElement[]>((resolve, reject) => {
+    const allPromise = new CancelablePromise(Promise.all([getDetailPromise, getGwPromise, getPeerAuthsPromise]));
```

Comment:
I don't understand why you are using a CancelablePromise here when there is no possibility to cancel it (you are not using the `PromiseRegistry` or canceling directly the promise anywhere)

Reply:
Right, there was no need.  That's just an artifact of stealing code and not properly cleaning it up.